### PR TITLE
Detect new Virtualization framework under macOS

### PR DESCRIPTION
**Pull request by chloerei**
Context: https://github.com/EugenMayer/docker-sync/issues/799

Check `com.docker.virtualization` process name in `.docker_for_mac?` for new Virtualization framework. Which effect default sync strategy in mac. Without this patch, docker-sync will use unison instead of native_osx.

I use in my project, looks work:

```
$ docker-sync start
          ok  Starting native_osx for sync geeknote-sync
geeknote-sync
unison: stopped
unison: started
     success  Sync container started
     success  Starting Docker-Sync in the background
```

Edit some file:

```
$ docker logs -f geeknote-sync
Looking for changes
Reconciling changes
changed  ---->            app/views/posts/index.html.erb  
Propagating updates
UNISON 2.51.3 (OCAML 4.12.0) started propagating changes at 15:40:16.22 on 06 May 2022
[BGN] Updating file app/views/posts/index.html.erb from /host_sync to /app_sync
[END] Updating file app/views/posts/index.html.erb
UNISON 2.51.3 (OCAML 4.12.0) finished propagating changes at 15:40:16.26 on 06 May 2022
Saving synchronizer state
Synchronization complete at 15:40:16  (1 item transferred, 0 skipped, 0 failed)
Looking for changes
Reconciling changes
Nothing to do: replicas have not changed since last sync.
```

But I'm not very clear about the internals of docker-sync, maybe need more test that native_osx strategy works fine on new Virtualization framework.

### CODE REVIEW

**Comment by EugenMayer** [comment]
Thank you for your contribution!

> But I'm not very clear about the internals of docker-sync, maybe need more test that native_osx strategy works fine on new Virtualization framework.

Are you using the new virtualization framework on your device and did the detection work for you? If yes, that's great.

Can you easily switch to the old variant and test, if that is still detected

File: spec/lib/docker-sync/dependencies/docker_driver_spec.rb [hunk 1]
```diff
@@ -25,6 +25,15 @@
...
     end
 
+    it 'checks if Docker is running in Virtualization' do
+      allow(DockerSync::Environment).to receive(:system).with('pgrep -q com.docker.hyperkit').and_return(false)
+      allow(DockerSync::Environment).to receive(:system).with('pgrep -q com.docker.virtualization').and_return(true)
+
+      is_expected.to be true
+      expect(DockerSync::Environment).to have_received(:system).with('pgrep -q com.docker.hyperkit')
+      expect(DockerSync::Environment).to have_received(:system).with('pgrep -q com.docker.virtualization')
+    end
```
Comment (EugenMayer) [inline spec/lib/docker-sync/dependencies/docker_driver_spec.rb:35]: Great you adopted the test!

File: lib/docker-sync/dependencies/docker_driver.rb [hunk 2]
```diff
@@ -5,7 +5,7 @@ module Driver
         def self.docker_for_mac?
           return false unless Environment.mac?
           return @docker_for_mac if defined? @docker_for_mac
-          @docker_for_mac = Environment.system('pgrep -q com.docker.hyperkit')
```
Comment (EugenMayer) [inline lib/docker-sync/dependencies/docker_driver.rb:8]: Let's add a comment what stands for what `com.docker.hyperkit` for the 'old virtualization' and `com.docker.virtualization` for the new one (maybe a link?)

**Review by EugenMayer — CHANGES_REQUESTED** [review]

**Comment by chloerei** [comment]
> Can you easily switch to the old variant and test, if that is still detected

Yes it still work.

File: lib/docker-sync/dependencies/docker_driver.rb (hunk 2 above)
Comment (chloerei) [reply lib/docker-sync/dependencies/docker_driver.rb:8]: I know the new virtualization framework options from here https://www.docker.com/blog/speed-boost-achievement-unlocked-on-docker-desktop-4-6-for-mac/

But the performance improvement is not good enough for me, so I still need docker-sync for big projects.

Maybe it's better for you to write the comment, English is not my native language.

Comment (EugenMayer) [reply lib/docker-sync/dependencies/docker_driver.rb:8]: English is not my native tounge either

i would suggest

`Detects new docker-desktop-for-mac virtualization engine, see https://www.docker.com/blog/speed-boost-achievement-unlocked-on-docker-desktop-4-6-for-mac/`

Comment (EugenMayer) [reply lib/docker-sync/dependencies/docker_driver.rb:8]: I would say IMOH, just use proper semver, and afaik v1.0 is considered equal to 1.0

**Pull request merged**